source/tools/glkernel-cli/PngExporter.h

```diff
@@ -0,0 +1,20 @@
+#pragma once
+
+#include "AbstractKernelExporter.h"
+
+#include <png.h>
+
+class PngExporter : public AbstractKernelExporter
+{
+public:
+    PngExporter(const cppexpose::Variant & kernel, const std::string & outFileName) :
+        AbstractKernelExporter{kernel, outFileName} {}
+
+    void exportKernel() override;
+
+protected:
+    template <typename T>
+    png_bytepp toPng(const glkernel::tkernel<T> & kernel, const int channels);
+
+    void writeToFile(png_bytepp data, const int colorType, const png_uint_32 height, const png_uint_32 width);
```

Comment:
Constness of primitive types are not part of the interface/declaration of a function. You should add them only for the definition.